plugins/lua/logistics.lua

```diff
@@ -49,16 +51,34 @@ local function print_stockpile_data(data)
 
     print('Designated/designatable items in stockpiles:')
     print()
-    local fmt = '%6s  %-' .. name_len .. 's  %4s %10s  %5s %11s  %4s %10s  %5s %11s';
-    print(fmt:format('number', 'name', 'melt', 'melt items', 'trade', 'trade items', 'dump', 'dump items', 'train', 'train items'))
+    --          sp.#  __sp.name____________  __melt__  _trade__  __dump__  _train__  frbd  un/frbd
+    local fmt = '%6s  %-' .. name_len .. 's  %4s %10s  %5s %11s  %4s %10s  %6s %11s  %12s  %12s';
+    print(fmt:format('number', 'name',
+            'melt',  'melt items',
+            'trade', 'trade items',
+            'dump',  'dump items',
+            'train', 'train items',
+            'forbid items',
+            '[un][forbid]'))
```

Comment:
why are these in the opposite order from the other columns?

Reply:
I played with this too and the result came from,
1. Showing un-forbidden items / un-forbidden-able items is basically useless 
2. Showing stockpiles with auto-unforbid enabled is useful, same for auto-forbid
3. Showing forbidden / forbid-able items is semi-valuable
5. Space is limited… 

So to get all the underlined headers on a single line but also add all 4 new columns would’ve been low value and too many characters using up the limited horizontal space. A single [un][forbid] header was space conscious and I believed was appropriate since both can’t be enabled simultaneously. To my eye it looked better having both of those to the right of ‘forbid items’ - are you suggesting it be to the left of ‘forbid items’? I can try that.

Other thoughts when I was doing this, my [setting] concept was born out of trying to have a single heading for two checkboxes, perhaps it would be worth the horizontal space to make other headers follow that format?
[melt] [trade] [dump] [train] [un][forbid]

Or perhaps I’ve misunderstood your concern? I’ll post some comparison screenshots in discord of some options.

Reply:
I was fine with the column contents - I was just curious about the order. Other columns have the checkboxes before the numbers instead of after

Reply:
Ah, I see. I'll try that.

Reply:
![image](https://github.com/DFHack/dfhack/assets/150195674/1f4ac079-93ca-4875-879d-ead87923d53f)
![image](https://github.com/DFHack/dfhack/assets/150195674/92a13228-37a0-4076-96f1-f607129ad7d2)

So I remember why I changed the order now, but I don't like my reasoning. But since it was your actual question: 'forbid items' causes the horizontal distance between checkboxes and item counts to be larger than any other section and it starts to look like it is more associated with training than forbidden items. At one point I reversed the columns and removed the 'forbid' from 'forbid items' hoping to effectively label both 'forbid' and save on that horizontal space.

Thoughts on something closer to this?
![image](https://github.com/DFHack/dfhack/assets/150195674/1b28ace9-f6f3-442b-bdf3-2864f1e34ce4)

Reply:
some of the spacing still needs tweaking in that last image, but curious for your thoughts on the concept which should be clear enough

Reply:
looks good to me

Reply:
> edit: trade and train are both t -- maybe make the letter match the hotkey in the overlay?
You read my mind! I was going to ask what a good way to grab the user's configured hotkey was so I could do exactly that (in addition to matching the overlay, but I suppose that gets risky?). Any suggestion or example to point me towards? In the meantime I made it 'a' matching the existing overlay.

Meanwhile I haven't been able to stop tweaking it... but I think I'm basically done with the look/layout pending a few bugs to tackle and any further feedback:
![image](https://github.com/DFHack/dfhack/assets/150195674/0ed059ab-b22e-4d71-a4cb-ca26058686c0)
See anything else you dislike or would prefer improved?

Reply:
Oh, that example doesn't have the capital [M] for masterworks melting enabled stockpile example, but that's there for both the melt column display and note (which only displays when it exists, per previous code).

Reply:
This is as additive as the game! Once again I couldn't leave the couple nitpicks I saw alone in the prior screenshot, so here's a refreshed version, 
![image](https://github.com/DFHack/dfhack/assets/150195674/90fe4d90-2127-4ab6-b32d-d9a02c157e64)
Looking forward to any feedback!

Definitely going to work now! lol

Reply:
> This is as additive as the game!
> Definitely going to work now! lol

welcome to my life : )

Reply:
> a good way to grab the user's configured hotkey

You certainly *can* get the key that a keybinding resolves to (follow the code that displays hotkey hints to see how), but most DFHack hotkeys use `CUSTOM_<foo>` keybindings, which are not usually changed by the player. E.g. the hotkey for training is `CUSTOM_CTRL_A`, so it's perfectly reasonable for you to just use `a` as the marker

Reply:
> it's perfectly reasonable for you to just use `a` as the marker
I’ll leave my concept of a hotkey matching on my wishful-thinking backlog and just match letters with the existing vanilla+dfhack experience (i.e. use `[a]` for `autotrain` enabled).

…but just to clarify the idea (maybe this is a case of a bad idea with problems that stimulates better idea)…

Concept: retain DFHack `CUSTOM_CTRL_<letter>` hotkey pattern for Stockpiles Overlay ToggleHotkeyLabels, BUT dynamically change its `<letter>` to correspond to the currently configured vanilla DF hotkey letter for that matching vanilla designation. This applies most directly to melt, dump, and forbid.
Example: So when Vanilla DF hotkey for Dump designations is configured for `[i, d]` then Stockpile Overlay would match the letter `d` for dump and configure its auto-dump toggle hotkey with `[ctrl+d]`. And this same letter (lowercase) would also be what is displayed here in the logistics status output to denote which stockpiles have auto-dump enabled (`[d]`). That way, the mental “muscle memory” of dump = ‘d’ applies in all three experiences:
a.) Vanilla DF dump designation 
b.) Stockpile Overlay auto-dump
c.) Text output denoting dump, such as logistics status output
And this like-for-like letter-to-function experience would hold true even when the user customizes their DF keys (which is a popular DF Workshop item as you likely know and grudgingly deal with). The actual hotkey combination ([i, d] vs [ctrl+d]) would not be the same, just the letter would match.
Obvious Risk: (and presumably deal breaker) A user could configure letters for these functions which in vanilla DF that conflicts with something else in one of these DFHack experiences. This likelihood would then necessitate hotkey fallback logic to a set of unique defaults when a conflict exists. This becomes particularly high maintenance (and even confusing to a user experimenting with changing a particular DF hotkey). Especially if this pattern was extended to additional DFHack experiences. So… dead concept.

Reply:
> Obvious Risk
yeah